extensions/skyportal/static/js/components/Archive.jsx

```diff
@@ -0,0 +1,541 @@
+import React, {Suspense, useEffect, useState} from "react";
+
+import Card from "@material-ui/core/Card";
+import CardActions from "@material-ui/core/CardActions";
+import CardContent from "@material-ui/core/CardContent";
+
+import Typography from "@material-ui/core/Typography";
+import TextField from "@material-ui/core/TextField";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+
+import Grid from "@material-ui/core/Grid";
+
+import Button from "@material-ui/core/Button";
+import {createMuiTheme, makeStyles, MuiThemeProvider, useTheme} from "@material-ui/core/styles";
+import { useForm, Controller } from "react-hook-form";
+import Paper from "@material-ui/core/Paper";
+import MUIDataTable from "mui-datatables";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+import {useDispatch, useSelector} from "react-redux";
+
+import TableRow from "@material-ui/core/TableRow";
+import TableCell from "@material-ui/core/TableCell";
+import {dec_to_dms, ra_to_hours} from "../units";
+import { showNotification } from "baselayer/components/Notifications";
+
+import * as Actions from "../ducks/archive";
+import SaveIcon from "@material-ui/icons/Save";
+import {IconButton} from "@material-ui/core";
+import HelpOutlineIcon from "@material-ui/icons/HelpOutline";
+import Popover from "@material-ui/core/Popover";
```

Comment:
Sorry, this is a little picky but would be nice if the imports were more organized here, i.e. all the MUI ones together and then all the other SP imports grouped together

Reply:
Yea, I've been treating js code as simply very long spells that eventually do magic once you've added the proper ingredient to the cauldron... Fixed.